### PR TITLE
Dimensions Computation Tutorial

### PR DESCRIPTION
**Pull request by chinmaykolhatkar**
This is doc is completed in review from Tim as Engineering Peer Reviewer and also from Amruta as Editorial Reviewer.

### CODE REVIEW

File: docs/operators/dimensioncomputation/dimension_computation_store.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
+#Dimensions Computation tutorial
+
+## About this document
+The *Dimensions Computation Tutorial* contains a brief introduction of the Dimension Computation usecase. This tutorial provides detailed steps for performing dimensions computation. Additionally, this tutorial contains instructions for using DataTorrent's out-of-the-box operators to easily add dimensions computation to an application.
+
+## Getting to know Dimensions Computations
+Big data scenarios often come with a problem -- the volumes of  of data flowing into big data systems are huge, and these volumes need to be observed for historical trends.
```
Comment (gauravgopi123) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: "the volumes of of" => "the volumes of"

File: docs/operators/dimensioncomputation/dimension_computation_store.md [hunk 2]
```diff
@@ -0,0 +1,368 @@
+#Dimensions Computation tutorial
+
+## About this document
+The *Dimensions Computation Tutorial* contains a brief introduction of the Dimension Computation usecase. This tutorial provides detailed steps for performing dimensions computation. Additionally, this tutorial contains instructions for using DataTorrent's out-of-the-box operators to easily add dimensions computation to an application.
```
Comment (gauravgopi123) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: Does it make sense to break the topics into list. Something like
The *Dimensions Computation Tutorial* covers following
1. Brief introduction of the Dimension Computation usecase
2. Detailed steps for performing dimensions computation
3. Instructions for using DataTorrent's out-of-the-box operators to easily add dimensions computation to an application

File: docs/operators/dimensioncomputation/dimension_computation_store.md (hunk 1 above)
Comment (gauravgopi123) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: "Big data scenarios often come with a problem" doesn't sound good.. It should be rephrased to something like "Big data scenarios often have usecase where huge volumes of data flowing through the big data systems need to be observed for historical trends"

File: docs/operators/dimensioncomputation/dimension_computation_store.md [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+
+## Getting to know Dimensions Computations
+Big data scenarios often come with a problem -- the volumes of  of data flowing into big data systems are huge, and these volumes need to be observed for historical trends.
+Big data applications will greatly benefit if they are equipped with the functionality of viewing  historical data aggregated across time buckets. The process of receiving individual events, aggregating them over a duration, and using parameters to observe trends  is called *Dimensions Computation*.
+
+
+## Overview
+
+### What is Dimensions Computation?
+Dimensions Computation is a powerful mechanism that allows for spotting trends in streaming data in real-time. This tutorial will cover the concepts behind Dimensions Computation and provide details on the process of performing Dimensions Computation. We will also show you how to use Data Torrent's out of the box operators to easily add Dimensions Computation to an application.
```
Comment (gauravgopi123) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: There is repetition of what is mentioned in "About this document"

File: docs/operators/dimensioncomputation/dimension_computation_store.md [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+
+#### DimensionsStore operator
+The DimensionsStore operator  is an Operator that performs transient and final aggregations on the data generated by the Dimensions Computations operator. It maintains the historical data for aggregates to ensure a meaningful  historical view.
+
+
+## Dimensions Computation use cases
+Consider the case of a digital advertising publisher who receives thousands of click events every second. The history of individual clicks and impressions doesn't divulge details about  users and the advertisements. A technique for deriving meaning out of such data is to observe the total number of clicks and impressions every second, minute, hour, and day. Such a technique might be  helpful for determining global trends in the advertising system, but may not provide enough granularity for  localized trends. For example, the total clicks and impressions over a duration might lack in usefulness, however, the total clicks and impressions for a particular advertiser, a particular geographical area, or a combination of the two can provide actionable insight.
```
Comment (gauravgopi123) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: extra spaces in following
"about  users"
"for  localized"

File: docs/operators/dimensioncomputation/dimension_computation_store.md [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+
+
+## Architecture
+
+Dimensions Computation requires 4 operators working in sync: Dimensions Computation, Dimensions Store, Query, and Query Result. Given [DAG](#dag) is for a complete Dimensions Computation application.
```
Comment (gauravgopi123) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: Should Operators name be "DimensionsComputation", "DimensionsStore", "QueryResult"

File: docs/operators/dimensioncomputation/dimension_computation_store.md [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+1. **dt.operator.QueryResult.numRetries**
+This property indicates the maximum number of times the QueryResult operator should retry sending data. This value is usually high.
+
+##### Attributes
+1. **dt.operator.DimensionsComputation.attr.PARTITIONER:**
+This attribute determines the number of  partitions for DimensionsComputation. Adding more partitions means data is  processed in parallel. If this attribute is not provided, a single partition is created. Refer to the [Partitioning](#partitioning) section for details on partitioning.
+1. **dt.operator.DimensionsComputation.attr.MEMORY_MB:**
+This attribute determines the  memory that should be assigned to the DimensionsComputations operator. If this attribute is not provided, the default value of  1 GB is used.
+1. **dt.operator.Store.attr.MEMORY_MB:**
+This attribute determines the memory that should be assigned the to DimensionsStore operator. If this attribute is not provided, the default value of 1 GB is used.
```
Comment (gauravgopi123) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: assigned the to => assigned to

**Comment by sashadt** [comment]
Tim, waiting for you to update based on feedback before merging.

File: docs/operators/dimensioncomputation/dimension_computation_store.md (hunk 1 above)
Comment (chinmaykolhatkar) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: Done.

File: docs/operators/dimensioncomputation/dimension_computation_store.md (hunk 2 above)
Comment (chinmaykolhatkar) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: These topic have set by Amruta and the document already had gone trough editorial review.

File: docs/operators/dimensioncomputation/dimension_computation_store.md (hunk 3 above)
Comment (chinmaykolhatkar) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: Again, editorial review is completed. As I understood, repetition is added by her when the last used phrase is old

File: docs/operators/dimensioncomputation/dimension_computation_store.md (hunk 4 above)
Comment (chinmaykolhatkar) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: Done.

File: docs/operators/dimensioncomputation/dimension_computation_store.md (hunk 5 above)
Comment (chinmaykolhatkar) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: Yes. Made the changes.

File: docs/operators/dimensioncomputation/dimension_computation_store.md (hunk 6 above)
Comment (chinmaykolhatkar) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: Done.

**Comment by chinmaykolhatkar** [comment]
Hi Sasha,

I've take care of most of the change as mentioned by gaurav. Few of the asks (related to language) were against the editorial review, hence not made.

Feel free to pull in the changes if you're happy.

-Chinmay.

File: docs/operators/dimensioncomputation/dimension_computation_store.md (hunk 2 above)
Comment (gauravgopi123) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: I am not saying change topic. I am saying break the paragraph into bullet points... I don't think that Editorial review being done means it is final

Comment (amrutaphansalker) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: This is more a "preface" section that serves as an introduction to what is to follow. We can think of modifying this structure, if everyone feels the pressing need to. 

The style is adopted from books; usually, prefaces and preliminary chapters precede the main content.

File: docs/operators/dimensioncomputation/dimension_computation_store.md (hunk 3 above)
Comment (amrutaphansalker) [inline docs/operators/dimensioncomputation/dimension_computation_store.md]: Those are separate sections; the first ones talk about the intent to the book. The topics that follow are an introduction to dimension computation itself.

**Pull request merged**